pom.xml

```diff
@@ -1,245 +1,293 @@
+                <version>3.1.1</version>
+                <executions>
+                    <execution>
+                        <id>copy-painless</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>copy</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <artifactItems>
+                        <artifactItem>
+                            <groupId>org.codelibs.elasticsearch.module</groupId>
+                            <artifactId>lang-painless</artifactId>
+                            <version>5.5.0</version>
+                            <type>jar</type>
+                            <overWrite>false</overWrite>
+                            <outputDirectory>${project.build.directory}/es/modules/lang-painless</outputDirectory>
+                            <destFileName>lang-painless.jar</destFileName>
+                        </artifactItem>
+                        <artifactItem>
+                            <groupId>org.ow2.asm</groupId>
+                            <artifactId>asm-debug-all</artifactId>
+                            <version>5.1</version>
+                            <type>jar</type>
+                            <overWrite>false</overWrite>
+                            <outputDirectory>${project.build.directory}/es/modules/lang-painless</outputDirectory>
+                            <destFileName>asm-debug-all.jar</destFileName>
+                        </artifactItem>
+                        <artifactItem>
+                            <groupId>org.antlr</groupId>
+                            <artifactId>antlr4-runtime</artifactId>
+                            <version>4.5.1-1</version>
+                            <type>jar</type>
+                            <overWrite>false</overWrite>
+                            <outputDirectory>${project.build.directory}/es/modules/lang-painless</outputDirectory>
+                            <destFileName>antlr4-runtime.jar</destFileName>
+                        </artifactItem>
+                    </artifactItems>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>2.6</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.1</version>
+                <configuration>
+                    <encoding>UTF-8</encoding>
+                </configuration>
+            </plugin>
+        </plugins>
+        <resources>
+            <resource>
+                <directory>${project.build.directory}/es/</directory>
+            </resource>
```

Comment:
Those three lines were added and look unrelated. Or rather, if anything, they look like they should be replacing the three lines below

Reply:
No that's correct, that the temp directory the jars are copied in to (see line 229 for example) which then in turn is used by the resources plugin.